workers/ohsome_quality_analyst/api/api.py

```diff
@@ -122,74 +117,40 @@ def empty_api_response() -> dict:
 
 
 @app.get("/indicator")
-async def get_indicator(
-    name: IndicatorEnum,
-    layerName: LayerEnum,
-    includeSvg: bool = False,
-    bpolys: Optional[str] = None,
-    dataset: Optional[DatasetEnum] = None,
-    featureId: Optional[str] = None,
-    fidField: Optional[FidFieldEnum] = None,
-):
-    """Create an Indicator.
+async def get_indicator(parameters=Depends(IndicatorGETRequestModel)):
+    """Get an already calculated Indicator for region defined by OQT.
 
-    Either the parameters `dataset` and `featureId` have to be provided
-    or the parameter `bpolys` in form of a GeoJSON.
-
-    Depending on the input, the output is a GeoJSON Feature or
-    FeatureCollection with the indicator results.
-    The Feature properties of the input GeoJSON will be preserved
-    if they do not collide with the properties set by OQT.
+    The response is a GeoJSON Feature with the indicator results as properties.
+    To request an Indicator for a custom AOI pease use the POST method.
     """
-    if bpolys is not None:
-        bpolys = json.loads(bpolys)
-    raw = {
-        "name": name,
-        "layerName": layerName,
-        "includeSvg": includeSvg,
-        "bpolys": bpolys,
-        "dataset": dataset,
-        "featureId": featureId,
-        "fidField": fidField,
-    }
-    parameters = {k: v for k, v in raw.items() if v is not None}
     return await _fetch_indicator(parameters)
 
 
 @app.post("/indicator")
-async def post_indicator(
-    parameters: IndicatorRequestModel,
-):
+async def post_indicator(parameters: IndicatorPOSTRequestModel):
     """Create an Indicator.
 
-    Either the parameters `dataset` and `featureId` have to be provided
-    or the parameter `bpolys` in form of a GeoJSON.
-
-    Depending on the input, the output is a GeoJSON Feature or
-    FeatureCollection with the indicator results.
-    The Feature properties of the input GeoJSON will be preserved
+    Depending on the input, the output is a GeoJSON Feature or FeatureCollection with
+    the indicator results. The Feature properties of the input GeoJSON will be preserved
     if they do not collide with the properties set by OQT.
     """
     return await _fetch_indicator(parameters)
 
 
-@pydantic.validate_arguments
```

Comment:
where are the arguments now checked?

Reply:
I thought this to be necessary because I though FastAPI does not support using pydantic models with GET request.
But they support it.
`async def get_indicator(parameters=Depends(IndicatorGETRequestModel)):`
The arguments are checked with the function call to `get_indicator` or `post_indicator`.